resources/ts/contexts/DegreeProgramValidationProvider.tsx

```diff
@@ -1,169 +1,192 @@
-            }),
-        );
-    };
-
-    const fieldHasErrors = useCallback(
-        (fieldId: DegreeProgramDataPaths) => {
-            const fieldErrors = getFieldErrors(fieldId);
-
-            return fieldErrors.size > 0;
-        },
-        [getFieldErrors],
-    );
-
-    const removeError = (fieldId: DegreeProgramDataPaths) => {
-        setErrors((prevErrors) => {
-            const newErrors = new Map(prevErrors);
-            newErrors.delete(fieldId);
-            newErrors.delete(`${fieldId}.id` as DegreeProgramDataPaths);
-            return newErrors;
-        });
-    };
-
-    subscribe(() => {
-        const { isSavingPost, isAutosavingPost } = select(editorStore);
-        setIsSaving(isSavingPost() && !isAutosavingPost());
-    });
-
-    // Update server errors when post is saved.
-    useEffect(() => {
-        if (isSaving) {
-            return;
-        }
-
-        if (!postId) {
-            return;
-        }
-
-        const serverErrors = select(coreStore).getLastEntitySaveError(
-            'postType',
-            serverData().postType,
-            postId,
-        );
-
-        if (!serverErrors) {
-            setErrors(new Map());
-            return;
-        }
-
-        /**
-         * This is a workaround for inconsistent block editor behavior.
-         * The "Publish" button changes the post status, while the "Save draft" button doesn't.
-         * But if you tried to publish the post and got an error,
-         * and then click "Save draft", the post is published.
-         *
-         * @link https://github.com/WordPress/gutenberg/blob/fedf426e8bf6f32ca6ef0b60cf26ecc81f917e03/packages/editor/src/components/post-publish-button/index.js#L256
-         * @link https://github.com/WordPress/gutenberg/blob/fedf426e8bf6f32ca6ef0b60cf26ecc81f917e03/packages/editor/src/components/post-saved-state/index.js#L153
-         */
-        editPost({ status: 'draft' }, { undoIgnore: true });
-
-        setErrors(new Map([...transformServerError(serverErrors)]));
-    }, [isSaving, postId]);
-    return (
-        <DegreeProgramValidationContext.Provider
-            value={{ errors, getFieldErrors, removeError, fieldHasErrors }}
-        >
-            {children}
-        </DegreeProgramValidationContext.Provider>
-    );
+const DegreeProgramValidationProvider = ( { children }: Props ) => {
+	const [ errors, setErrors ] = useState< FormValidationErrors >( new Map() );
+	const [ isSaving, setIsSaving ] = useState< boolean >( false );
+	const postId = useSelect< MapSelect< EditorStoreSelectors > >(
+		( selectCb ) => {
+			const { getCurrentPostId } = selectCb( editorStore );
+
+			return getCurrentPostId();
+		},
+		[]
+	);
+
+	const { editPost } = useDispatch( editorStore );
+
+	/**
+	 * Get an array of errors associated with a field
+	 * For example If you pass `title` you'll also get errors
+	 * with `title.en` and `title.de` and any other subfield error
```

Comment:
Let's improve the wording as part of the Boy Scouts rule.
```suggestion
	 * Get an array of errors associated with a field.
	 * For example, `passing `title` returns errors from
	 * `title.en`, `title.de`, and any other subfield.
```
@amiut, is "errors from" instead of "errors with" factually correct?